hal/src/stm32f2xx/spi_hal.c

```diff
@@ -618,7 +624,22 @@ static void spiOnSelectedHandler(void *data) {
         }
     }
 
-    if (spiState[spi].on_select_user_callback) {
-        spiState[spi].on_select_user_callback(state);
+    if (spiState[spi].select_user_callback) {
+        spiState[spi].select_user_callback(state);
+    }
+}
+
+int hal_spi_sleep(hal_spi_interface_t spi, bool sleep, void* reserved) {
+    if (sleep) {
+        CHECK_TRUE(hal_spi_is_enabled(spi), SYSTEM_ERROR_NONE);
+        CHECK_FALSE(spiState[spi].suspended, SYSTEM_ERROR_NONE);
+        hal_spi_transfer_dma_cancel(spi);
```

Comment:
This seems different from Gen 3 implementation where we wait for the transfer to complete. Let's set on a single behavior between platforms.

Reply:
I'd argue that when going to sleep we should wait instead in order not to leave the remote peripheral with an unfinished transfer and inconsistent data.

Also endless loop. `dma_aborted` is normally `0` and `hal_spi_transfer_dma_cancel()` will not do anything if there is not an ongoing transfer.